### PR TITLE
feat: Sync commits from `aztec-packages`

### PR DESCRIPTION
**Pull request by AztecBot**
BEGIN_COMMIT_OVERRIDE
feat: Allow nested arrays and vectors in Brillig foreign calls (https://github.com/AztecProtocol/aztec-packages/pull/4478)
feat: allow brillig to read arrays directly from memory (https://github.com/AztecProtocol/aztec-packages/pull/4460)
feat(avm): back in avm context with macro - refactor context (https://github.com/AztecProtocol/aztec-packages/pull/4438)
chore!: rename bigint_neg into bigint_sub (https://github.com/AztecProtocol/aztec-packages/pull/4420)
chore: add bigint solver in ACVM and add a unit test for bigints in Noir (https://github.com/AztecProtocol/aztec-packages/pull/4415)
feat!: Add expression width into acir (https://github.com/AztecProtocol/aztec-packages/pull/4014)
feat: Add bit size to const opcode (https://github.com/AztecProtocol/aztec-packages/pull/4385)
feat(aztec-nr): initial work for aztec public vm macro (https://github.com/AztecProtocol/aztec-packages/pull/4400)
chore: surpress chained macro warning (https://github.com/AztecProtocol/aztec-packages/pull/4396)
feat!: init storage macro (https://github.com/AztecProtocol/aztec-packages/pull/4200)
chore(acir)!: Move `is_recursive` flag to be part of the circuit definition (https://github.com/AztecProtocol/aztec-packages/pull/4221)
END_COMMIT_OVERRIDE

### CODE REVIEW

File: noir_stdlib/src/bigint.nr [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    #[builtin(bigint_mul)]
+    pub fn bigint_mul(_self: Self, _other: BigInt) -> BigInt {
+    }
+    #[builtin(bigint_div)]
+    pub fn bigint_div(_self: Self, _other: BigInt) -> BigInt {
+    }
+    #[builtin(bigint_from_le_bytes)]
+    pub fn from_le_bytes(_bytes: [u8], _modulus: [u8]) -> BigInt {}
+    #[builtin(bigint_to_le_bytes)]
+    pub fn to_le_bytes(_self: Self) -> [u8] {}
```
Comment (TomAFrench) [inline noir_stdlib/src/bigint.nr:24]:
```suggestion
    #[builtin(bigint_add)]
    pub fn bigint_add(self, other: BigInt) -> BigInt {
    }
    #[builtin(bigint_neg)]
    pub fn bigint_neg(self, other: BigInt) -> BigInt {
    }
    #[builtin(bigint_mul)]
    pub fn bigint_mul(self, other: BigInt) -> BigInt {
    }
    #[builtin(bigint_div)]
    pub fn bigint_div(self, other: BigInt) -> BigInt {
    }
    #[builtin(bigint_from_le_bytes)]
    pub fn from_le_bytes(bytes: [u8], modulus: [u8]) -> BigInt {}
    #[builtin(bigint_to_le_bytes)]
    pub fn to_le_bytes(self) -> [u8] {}
```

Comment (sirasistant) [reply noir_stdlib/src/bigint.nr]: Won't this generate unused warnings in stdlib when we reenable the import for bigint?

Comment (TomAFrench) [reply noir_stdlib/src/bigint.nr]: https://github.com/noir-lang/noir/pull/4179

Comment (sirasistant) [reply noir_stdlib/src/bigint.nr]: nice!

**Comment by sirasistant** [comment]
(this PR needs a new bb version)

**Pull request merged**